### DISCUSSION TITLE
Zigbee router

### DISCUSSION THREAD

**Discussion by NuccioT** [indomus/forum#806] [ZigBee, Z-Wave, Thread, Matter]
Buongiorno a tutti,
ho appena acquistato due sonoff zigbee bridge pensando (probabilmente sbagliando) che per costruire una rete domestica ci fosse necessità di più di questi devices. Avevo interpretato che il primo facesse da coordinatore e gli altri da router. Leggendo la documentazione su internet apprendo che alcuni devices alimentati possono fungere da router. Il mio problema è che devo coprire un'area molto estesa e quindi ho bisogno di "rilanciare" il segnale in più zone (per esempio vorrei installare delle valvole termostatiche zigbee tuya). Abito in una casa a più piani con altri familiari per cui condividiamo alcuni servizi, per esempio la rete.
C'è qualcuno che può darmi qualche dritta?

**Comment by indomus** [comment]
Ok: comincia col leggere BENE questo link: https://indomus.it/componenti/zigbee/

**Reply by NuccioT** [reply]
Grazie mille. Credo che ciò che mi ha indotto ad acquistare due getaway derivi proprio da questo articolo:
"ZigBee Router (ZR): questi dispositivi agiscono come router intermedi passando i dati da e verso altri dispositivi. **Di fatto non vi sono distinzioni hardware tra un ZC e un ZR** se non che viene rilasciato al coordinator il ruolo di inizializzare la rete, dopodiché diventano dispositivi identici."
Ho interpretato il virgolettato come: se uno lo faccio diventare coordinator tutti gli altri a cascata diventano router.

**Reply by snakuzzo** [reply]
Diciamo che dipende anche dal tipo di HW che utilizzi.
Se usassi una CC2531/CC2530 etc. lo stesso HW potrebbe essere flashato con firmware zStack coordinator o router ed usarlo quindi in maniera diversa a seconda delle esigenze.
Con il Sonoff Bridge Zigbee non dovrebbe essere possibile, ma non ci metterei la mano sul fuoco.

EDIT: Forse la conferma arriva anche da [QUESTO THREAD](https://github.com/arendst/Tasmota/issues/9199) su Tasmota

**Reply by NuccioT** [reply]
I due sonoff sono flashati con Tasmota e per esigenze personali (è da nove mesi che sbatto la testa contro home assistant) li ho configurati come ZHA adottando l'integrazione in Hassio. Ho avuto la fortuna di settarli separatamente per cui solo uno per ora è stato integrato in HA. Ora il mio punto è: se devo coprire un'area abbastanza vasta, compreso un giardino dove vorrei piazzare dei sensori di movimento che non fanno da router, come ne esco?

**Reply by snakuzzo** [reply]
i device che ti fanno da router sono quelli alimentati NON a batteria...quelli a batteria non funzionano da router.
In base alle testimonianze a destra e a manca, potresti avere anche problemi di compatibilità tra device di "marche" diverse.
Usando invece un chip di Texas Instruments tipo CC2530 flashato con fiw zStack router non dovresti avere problemi

**Reply by NuccioT** [reply]
Grazie. Quindi seguendo il tuo consiglio dovrei dotarmi di un po' di dongle TI e flasharli con zStak: corretto?
Se invece, siccome le reti attive in casa sono più di una, collegassi due sonoff bridge come coordinator agganciandoli a nomi di reti diverse? Ovviamente tutte le reti partono dal medesimo router e non so se questo possa causare problemi.

**Reply by snakuzzo** [reply]
Puoi pure usare due coordinator, ma non è detto che risolvi i problemi di raggiungibilità...inoltre il fatto che colleghi due coordinator a reti WIFI diverse, non c'entra niente...anche se fossero su due classi IP diverse sarebbe la stessa cosa. Se usi ZHA puoi configurare un solo coordinator se non vado errato.
Usando invece zigbee2tasmota via MQTT e non con ZHA, potresti usare in contemporanea due coordinator (magari usando base topic diversi), ma dovresti configurare a mano tutte le entità mqtt e configurare ad hoc diversi SetOption per semplificarti la vita

**Reply by NuccioT** [reply]
ok, grazie. ho provveduto ad acquistare qualche sonoff basiczb e sto verificando quanti mini mi servono. Mettendo questi che sono collegati alla rete il segnale viene rilanciato vero? Buon fine settimana

**Reply by dompa72** [reply]
i sonoff zbmini rilanciano anche a rete

come coordinator ho questa https://www.tindie.com/products/gio_dot/z-bee-duo-hybrid-cc2652p-zigbee-30-adapter/ 
considerando che è wifi posso installarla nella retee funziona anche quando riavvio l'host di ha

**Reply by snakuzzo** [reply]
Come già detto da @dompa72, i sonoff zb mini, essendo device zigbee alimentati NON a batteria, fungono anche da router.
Non è sicuro che fungano da router per qualsiasi device...te lo accennavo l'altro giorno [QUI](https://github.com/indomus/forum/discussions/806#discussioncomment-1413972)

@dompa72 anche io uso la Z-Bee Duo di giovanni con modulino ESP-01 per slegarla completamente da altri device ed usarla con ser2net. Credo che tra le tante soluzioni attualmente in commercio (in attesa di provare la nuova chiavetta di ITEAD, anche se non ha un chip integrato per ser2net), sia quella migliore

**Comment by D0nat09** [comment]
Ciao, ci sono così tante innovazioni al giorno d'oggi che ne sono un po' scioccato. Ma l'unica cosa è che non tutti sono di buona qualità. Inoltre, è molto frustrante che non ho avuto un buon sistema di allarme la prima volta. Ma recentemente ho trovato un sistema ajax che non solo aiuta a scoraggiare i ladri, ma anche a trovare ilevatore di perdite, e a riparare le inondazioni. Comunque, è tutto bello e funziona davvero. Ora posso semplicemente andare nell'app e controllare lo stato del mio appartamento. E se succede qualcosa, ricevo subito un avviso sul mio portachiavi.
